### PR TITLE
Refactor FXIOS-4688 [v105] Extracted Wallpaper URL to test it

### PR DESCRIPTION
**Pull request by electricRGB**
[JIRA Reference](https://mozilla-hub.atlassian.net/browse/FXIOS-4688)

### CODE REVIEW

**Comment by electricRGB** [comment]
@yoanarios I added you as a reviewer as well, because of unit testing. If there some other unit testing you think I should add, please feel free to reach out and let me know - I was thinking specifically about, maybe fake passing in a bundle? to get the test url from there? So I'm still testing the bundle fetching aspect too. I'm always learning.

File: Client/Frontend/Home/Wallpapers/v2/Utilities/WallpaperURLProvider.swift [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+enum WallpaperURLType {
+    case metadata
+    case imageURL
+}
+
+struct WallpaperURLProvider {
+    // MARK: - Properties
+    enum URLProviderError: Error {
+        case noBundledURL
+        case problemBuildingURL
```
Comment (thatswinnie) [inline Client/Frontend/Home/Wallpapers/v2/Utilities/WallpaperURLProvider.swift:17]: Houston we have a problem? What do you think about renaming it to something like `invalidURL` or  just `urlBuilding`?

Comment (electricRGB) [reply Client/Frontend/Home/Wallpapers/v2/Utilities/WallpaperURLProvider.swift:17]: sure.

**Review by thatswinnie — APPROVED** [review]

**Review by yoanarios — APPROVED** [review]

**Comment by electricRGB** [comment]
BR was green before a trivial push. Won't waste credits re-running an insignificant change.

Screen Shot 2022-08-03 at 10 17 38 AM

**Pull request merged**